assets/js/src/settings/pages/woo_commerce/checkout_optin.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { t, onToggle, onChange } from 'common/functions';
+import { Label, Inputs } from 'settings/components';
+import { useSetting, useAction } from 'settings/store/hooks';
+
+export default function CheckoutOptin() {
+  const [enabled, setEnabled] = useSetting('woocommerce', 'optin_on_checkout', 'enabled');
+  const [message, setMessage] = useSetting('woocommerce', 'optin_on_checkout', 'message');
+  const setErrorFlag = useAction('setErrorFlag');
+  const emptyMessage = message.trim() === '';
+  React.useEffect(() => {
+    setErrorFlag(emptyMessage);
+  }, [emptyMessage, setErrorFlag]);
+
+  return (
+    <>
+      <Label
+        title={t('wcOptinTitle')}
+        description={t('wcOptinDescription')}
+        htmlFor="mailpoet_wc_checkout_optin"
+      />
+      <Inputs>
+        <input
+          type="checkbox"
+          id="mailpoet_wc_checkout_optin"
+          data-automation-id="mailpoet_wc_checkout_optin"
+          checked={enabled === '1'}
+          onChange={onToggle(setEnabled, '')}
+        />
+      </Inputs>
+      {enabled === '1' && (
+        <>
+          <Label
+            title={t('wcOptinMsgTitle')}
+            description={t('wcOptinMsgDescription')}
+            htmlFor="mailpoet_wc_checkout_optin_message"
+          />
+          <Inputs>
+            <input
```

Comment:
This input either needs `size="48"` or `class="regular-text"` so it's not so small.